### DISCUSSION TITLE
Pull request to release

### DISCUSSION THREAD

**Discussion by JakeGinnivan** [nrwl/nx#28157] [Feature Requests]
The majority of my projects use https://github.com/changesets/action as the release process.

What this action does is if there planned changes (ie, file based versioning, and changeset markdown files in the changeset repo), it will open a 'Release' pull request, the description contains the unreleased changes all listed.

As more version plans are added, the PR just gets updated with the running to be released changes. Here is a screenshot

![image](https://github.com/user-attachments/assets/b06f688b-ea74-4f67-9da7-04552b54850f)

When the PR is merged, the action will see there are no planned changes (the PR updates the changelog.md, and version numbers in package.json files and deletes the planned files).

Because there are no changes it will run the `release` command instead of the `version` command. This releases the packages, creates github releases etc.

Having the PR as a release trigger is great, because you can immediately see what are the unreleased changes in any repo. 

A quick look into the release process of NX has the versioning step (merging the plan files into the  changelog files and updating version numbers) is done within the release step.

The other reason having a pull request is great is for branch protections. The pull request for releasing contains all git changes, so the merge process updates all the files in git, then the release process creates the github release and publishes packages etc.

How could we have a similar setup with NX release?

**Comment by sir-captainmorgan21** [comment]
Hey,  @JakeGinnivan have you made any progress around this? Im looking to do something similar. Essentially following the release-please package from google.

**Reply by JakeGinnivan** [reply]
I haven't, still would like to do it but changesets it doing the job for the moment until I get time.
